src/core/common/message.cpp

```diff
@@ -41,11 +41,12 @@
 #include "net/checksum.hpp"
 #include "net/ip6.hpp"
 
+#if OPENTHREAD_MTD || OPENTHREAD_FTD
```

Comment:
Do we need this check? Is it required for heap use?

Reply:
No, just moved the check from `otError MessagePool::ReclaimBuffers(int aNumBuffers, Message::Priority aPriority)` here. Because Message is not used at all on RCP.